llumnix/backends/vllm/llm_engine.py

```diff
@@ -238,22 +242,25 @@ def restart_workers(self) -> None:
     def add_request(self,
                     request_id: str,
                     server_info: ServerInfo,
+                    req_expected_step: int,
                     *args,
                     **kwargs) -> None:
         # When manager is unavailable, api server might dispatch the request that has already been dispatched.
         if request_id in self.engine.request_server_info:
             return
         # Store the server information of each request to put the request outputs back to the corresponding api server correctly.
         self.engine.request_server_info[request_id] = server_info
+        with self.engine.scheduler.scheduler_lock: 
```

Comment:
why not use the `@scheduler_lock` decorator for a method to add a request info? That would be more consistent with the current code style.